presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -174,15 +178,20 @@ private DistributedQueryRunner(
             extraCoordinatorProperties.put("experimental.iterative-optimizer-enabled", "true");
             extraCoordinatorProperties.putAll(extraProperties);
             extraCoordinatorProperties.putAll(coordinatorProperties);
-            coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
-            servers.add(coordinator);
+            for (int i = 0; i < coordinatorCount; i++) {
+                TestingPrestoServer coordinator = closer.register(createTestingPrestoServer(discoveryUrl, false, resourceManagerEnabled, true, extraCoordinatorProperties, parserOptions, environment, baseDataDir));
+                servers.add(coordinator);
+                coordinators.add(coordinator);
+                extraCoordinatorProperties.remove("http-server.http.port");
```

Comment:
didn't quite understand why we want to remove this property?

Reply:
We want just one coordinator to use the specified port, as each one is running in the same host, and multiple processes can't share the same port.